### PR TITLE
Add optional comments reference_id

### PR DESCRIPTION
**Pull request by nickvergessen**
So for talks use of the comments api, we would be interested in an additional field for comments to be added. We would store there a random hash, so when the server messages are returned while you e.g. post a message we can see if the message is the one you are currently sending and already hide the temporary message.
You can currently see this sometimes when a lot of notifications are sent, that the message is shown twice for some milliseconds, because currently the temporary message is killed by the response of the post (which happens after the notifications are sent to the proxy) but in parallel the message appears already from the "waiting for new messages" request immediately after its written to the database.

So this PR adds the optional column, and only writes/updates it when the column was added already. This laziness is okay, as the impact is sloppier behaviour instead of broken behaviour in case the column was not added.

### CODE REVIEW

File: core/Application.php [hunk 1]
```diff
@@ -168,6 +169,23 @@ function (GenericEvent $event) use ($container) {
...
 
+		$eventDispatcher->addListener(IDBConnection::CHECK_MISSING_COLUMNS_EVENT,
+			function (GenericEvent $event) use ($container) {
+				/** @var MissingColumnInformation $subject */
+				$subject = $event->getSubject();
+
+				$schema = new SchemaWrapper($container->query(IDBConnection::class));
+
+				if ($schema->hasTable('share')) {
+					$table = $schema->getTable('share');
```
Comment (rullzer) [inline core/Application.php]: share?

File: apps/settings/lib/Controller/CheckSetupController.php [hunk 2]
```diff
@@ -445,6 +446,15 @@ protected function hasMissingIndexes(): array {
 		return $indexInfo->getListOfMissingIndexes();
 	}
 
+	protected function hasMissingColumns(): array {
+		$indexInfo = new MissingColumnInformation();
+		// Dispatch event so apps can also hint for pending index updates if needed
+		$event = new GenericEvent($indexInfo);
```
Comment (nickvergessen) [inline apps/settings/lib/Controller/CheckSetupController.php]: For now i copied the indexes code, but we could also go for a new event class here making things proper. i dont care too much

File: core/Application.php (hunk 1 above)
Comment (nickvergessen) [reply core/Application.php]: Right, that's why the check didn't work 😅

File: apps/settings/lib/Controller/CheckSetupController.php (hunk 2 above)
Comment (rullzer) [reply apps/settings/lib/Controller/CheckSetupController.php]: Depends. If we consider this public then I would go for the new things straight away

Comment (nickvergessen) [reply apps/settings/lib/Controller/CheckSetupController.php]: Okay, fine by me

**Review by rullzer — APPROVED** [review]
Fine by me

**Review by ChristophWurst — APPROVED** [review]

**Comment by nickvergessen** [comment]
Squash time

**Review by blizzz — APPROVED** [review]

**Comment by blizzz** [comment]
```
…
PhantomJS  2.1.1 (Linux 0.0.0) OC.SetupChecks tests checkSetup should return an  error if the protocol is https but the server generates http links  FAILED
TypeError: undefined is not an object (evaluating 'data.missingColumns.length') in core/js/setupchecks.js (line 1) 
afterCall@core/js/setupchecks.js:1:80437 
…
```

**Comment by nickvergessen** [comment]
Updated the one new test as well

**Comment by blizzz** [comment]
There's more:

```
…
There was 1 failure:
1) OCA\DAV\Tests\unit\Comments\CommentsNodeTest::testGetProperties 
Failed asserting that false is true. 

/drone/src/apps/dav/tests/unit/Comments/CommentsNodeTest.php:487 
```

i didn't check all steps.

**Pull request merged**